### PR TITLE
CURATOR-587 zookeeper embedded - preview

### PR DESCRIPTION
**Pull request by eolivelli**
This is a preview patch for when we will have ZK 3.7.0 and the official ZooKeeperServerEmbedded API.

At that time we won't need to hack the internals of ZK

### CODE REVIEW

**Comment by eolivelli** [comment]
my IDE reformatted part of the code, I will revert

File: pom.xml [hunk 1]
```diff
@@ -69,7 +69,7 @@
         <redirectTestOutputToFile>true</redirectTestOutputToFile>
 
         <!-- versions -->
-        <zookeeper-version>3.6.3</zookeeper-version>
+        <zookeeper-version>3.7.1-SNAPSHOT</zookeeper-version>
```
Comment (eolivelli) [inline pom.xml:72]: we have to move this to 3.7.1

in 3.7.1 we have fixed a few problems in ZK Embedded, and 3.8.0 is also better

@tisonkun

Comment (tisonkun) [reply pom.xml:72]: Since @kezhuw resolves CURATOR-596 by #426 I'd cc you here if you're interested in taking over this patch also.

Comment (eolivelli) [reply pom.xml]: I have rebased on current master

**Pull request closed without merging**